zomes/rea_economic_resource/lib/src/lib.rs

```diff
@@ -116,11 +127,22 @@ impl API for EconomicResourceZomePermissableDefault {
         // if the event is a transfer-like event, run the receiver's update first
         if let MaybeUndefined::Some(receiver_inventory) = &event.to_resource_inventoried_as {
             let inv_entry_hash: &EntryHash = receiver_inventory.as_ref();
-            resources_affected.push(handle_update_inventory_resource(
+            let (header_hash, resource_address, new_resource, prev_resource) = handle_update_inventory_resource(
                 &resource_entry_def_id,
                 &get_latest_header_hash(inv_entry_hash.clone())?,   // :TODO: temporal reduction here! Should error on mismatch and return latest valid ID
                 event.with_inventory_type(ResourceInventoryType::ReceivingInventory),
-            )?);
+            )?;
+            resources_affected.push((header_hash, resource_address, new_resource, prev_resource));
+            // update index if transfer or transfer_all_rights action
+            if (event.get_action() == "transfer" || event.get_action() == "transfer_all_rights") && new_resource.primary_accountable != prev_resource.primary_accountable {
+                let new_value = if let Some(val) = &new_resource.primary_accountable { vec![val.to_owned()] } else { vec![] };
+                let prev_value = if let Some(val) = &prev_resource.primary_accountable { vec![val.to_owned()] } else { vec![] };
+                let e = update_index!(
+                    economic_resource
+                        .primary_accountable(new_value.as_slice())
+                        .not(prev_value.as_slice()));
+                hdk::prelude::debug!("update_economic_resource::primary_accountable index {:?}", e);
+            }
         }
         // after receiver, run provider. This entry data will be returned in the response.
         if let MaybeUndefined::Some(provider_inventory) = &event.resource_inventoried_as {
```

Comment:
let's add to this code block, to be robust, the same thing as you did above:

instead of this: 
hdk::prelude::debug!("update_economic_resource::to_resource_inventoried_as::primary_accountable index {:?}", e);
it would be:
hdk::prelude::debug!("update_economic_resource::resource_inventoried_as::primary_accountable index {:?}", e);

in theory (but maybe not in practice) both could be modified